src/main/kotlin/org/jetbrains/teamcity/github/util/WebHooksHelper.kt

```diff
@@ -0,0 +1,48 @@
+package org.jetbrains.teamcity.github.util
+
+import jetbrains.buildServer.serverSide.ProjectManager
+import jetbrains.buildServer.vcs.ChangesCheckingService
+import jetbrains.buildServer.vcs.OperationRequestor
+import jetbrains.buildServer.vcs.VcsRootInstance
+import org.eclipse.egit.github.core.Repository
+
+class WebHooksHelper(private val projectManager: ProjectManager, private val changesCheckingService: ChangesCheckingService) {
+
+    fun findRelevantVcsRootInstances(repository: Repository): Collection<VcsRootInstance> {
+        if (repository.cloneUrl.isNullOrEmpty())
+            return emptyList()
+
+        val repoUrls = setOf(repository.gitUrl, repository.cloneUrl, repository.sshUrl).filterNotNull().map { normalizeGitUrl(it) }
+
+        val prefilteredVcsRoots = projectManager.allVcsRoots
+            .filter {vcsRoot ->
+                isParamRefOrMatches(vcsRoot.vcsName) {
+                    it == "jetbrains.git"
+                }
+                && isParamRefOrMatches(normalizeGitUrl(vcsRoot.getProperty("url"))) {
+                    repoUrls.contains(it)
+                }
+            }
+        return prefilteredVcsRoots
```

Comment:
It seems this filter does not add versioned settings VCS roots to the result, see VcsRootInstanceFinder:402 in REST API

Reply:
Yes! Thank you! It seems we have to fix it in the GitHub App plugin too as it uses similar logic.

Reply:
Fixed now